vignettes/how-to-bench.Rmd

```diff
@@ -0,0 +1,225 @@
+
+Please note that even though many resources would only contain `source` and `target` columns (e.g. GO term:Gene, or TF:target (in this example)), the decouple wrapper requires `mor` and `likelihood` columns. As such, depending on the source set, arbitrary columns with these names should be provided. The `likelihood` column can be used to provide weights to each of the source:target interactions. Also, note that "tf", "target", and "confidence" can take any user-defined name (defined in the input tibble), where "confidence" is the column by which we wish to filter.
+
+In this tutorial, we will use [DoRothEA](https://github.com/saezlab/dorothea) - a gene set resource containing signed TF-target interactions.
+```{r source_example}
+readRDS_helper(source_url, T) %>%
+  slice_head(n=10) # show only first 10 rows
+```
+
+# Benchmark run
+Now that we have checked the prerequisites and we are confident that they are formatted appropriately, we can proceed with running the decoupleR benchmark pipeline.
+
+## Benchmark Settings and Assumptions
+Here we run the benchmark pipeline with DoRoThEA using the default settings.
+
+Note that the benchmark pipeline currently evaluates precision and that here we define the 'Condition positive' as the perturbed targets for each condition/experiment.
+Then we make the assumption that 'True positives (TPs)' would get a 'score' from each statistical method that would rank them within the top `n` scores, where `n` = the number of experiments covered by the set resource. On the other hand, a 'True Negative (TN)' is defined as a TF not perturbed and not ranked within the top `n` of TFs.
+'Condition Negatives' are defined in two ways depending whether downsampling is performed. 
+By defult, Condition Negatives are defined as the set of any TF regulon not perturbed in the current experiment. As such, each of these non-perturbed TF sets is appended and considered a Condition negative. This results in Condition Negatives vastly outnumbering  (equal to the number of non-perturbed TFs for all of the experiments) of Condition positives (equal to the number of truly perturbed target in each experiment).
+When downsampling is performed, the number of Condition Negatives is set to be the same as the number of Condition Positives (i.e. number of experiments covered by the gene set).
+
+In some cases, assuming that the perturbed target (in this case a TF) in a given experiment would be the most deregulated one (i.e. the one with highest `score`) is a flawed assumption due to the nature of regulation in biology. For example, a TF correlated or downstream of the perturbed target might be more perturbed than the experiment target itself. However, at this stage, we believe that this is likely our best choice. Nonetheless, we are open to suggestions and implementations of alternative source set performance methods.
+
+```{r bench_run, cache=TRUE, eval=TRUE, echo=T, results='hide', warning=FALSE}
```

Comment:
Cache is good when running interactively. Consider removing the file once the vignette finish. c: